tests/Fixer/FunctionNotation/NullableTypeHintForDefaultNullValueFixerTest.php

```diff
@@ -0,0 +1,240 @@
+{
+    /**
+     * @param string $expected
+     *
+     * @dataProvider provideDoNotFixCases
+     */
+    public function testDoNotFix($expected)
+    {
+        $this->doTest($expected);
+    }
+
+    public function provideDoNotFixCases()
+    {
+        yield ['<?php function foo($param = null) {}'];
+        yield ['<?php function foo($param1 = null, $param2 = null) {}'];
+        yield ['<?php function foo(&$param = null) {}'];
+        yield ['<?php function foo(& $param = null) {}'];
+        yield ['<?php function foo(/**int*/ $param = null) {}'];
+        yield ['<?php function foo(/**int*/ &$param = null) {}'];
+        yield ['<?php function foo(&/*comment*/$param = null) {}'];
+        yield ['<?php $foo = function ($param = null) {};'];
+        yield ['<?php $foo = function (&$param = null) {};'];
+
+        yield ['<?php function foo(?string $param = null) {}'];
+        yield ['<?php function foo(?string $param1 = null, ?string $param2 = null) {}'];
+        yield ['<?php function foo(?string &$param = null) {}'];
+        yield ['<?php function foo(?string & $param = null) {}'];
+        yield ['<?php function foo(?string /*comment*/$param = null) {}'];
+        yield ['<?php function foo(?string /*comment*/&$param = null) {}'];
+        yield ['<?php function foo(?string &/*comment*/$param = null) {}'];
+        yield ['<?php $foo = function (?string $param = null) {};'];
+        yield ['<?php $foo = function (?string &$param = null) {};'];
+
+        yield ['<?php function foo(?Baz $param = null) {}'];
+        yield ['<?php function foo(?\Baz $param = null) {}'];
+        yield ['<?php function foo(?Bar\Baz $param = null) {}'];
+        yield ['<?php function foo(?\Bar\Baz $param = null) {}'];
+        yield ['<?php function foo(?Baz &$param = null) {}'];
+        yield ['<?php function foo(?\Baz &$param = null) {}'];
+        yield ['<?php function foo(?Bar\Baz &$param = null) {}'];
+        yield ['<?php function foo(?\Bar\Baz &$param = null) {}'];
+        yield ['<?php function foo(?Baz & $param = null) {}'];
+        yield ['<?php function foo(?\Baz & $param = null) {}'];
+        yield ['<?php function foo(?Bar\Baz & $param = null) {}'];
+        yield ['<?php function foo(?\Bar\Baz & $param = null) {}'];
+        yield ['<?php function foo(?array &$param = null) {}'];
+        yield ['<?php function foo(?array & $param = null) {}'];
+        yield ['<?php function foo(?callable &$param = null) {}'];
+        yield ['<?php function foo(?callable & $param = null) {}'];
+        yield ['<?php $foo = function (?Baz $param = null) {};'];
+        yield ['<?php $foo = function (?Baz &$param = null) {};'];
+        yield ['<?php $foo = function (?Baz & $param = null) {};'];
+        yield ['<?php class Test { public function foo(?Bar\Baz $param = null) {} }'];
+        yield ['<?php class Test { public function foo(?self $param = null) {} }'];
```

Comment:
Please add test cases with comments and/or whitespaces between `?` and the type.

Reply:
Done